src/QsCompiler/SyntaxProcessor/CapabilityInference/ResultAnalyzer.fs

```diff
@@ -42,30 +42,55 @@ let createPattern kind range =
     let capability = TargetCapability.withResultOpacity opacity TargetCapability.bottom
 
     let diagnose (target: Target) =
-        let range = QsNullable.defaultValue Range.Zero range
-
-        match kind with
-        | _ when TargetCapability.subsumes target.Capability capability -> None
-        | ConditionalEqualityInOperation
-        | UnrestrictedEquality ->
-            let code =
+        if target.UseCapabilityWarnings then
+            let range = QsNullable.defaultValue Range.Zero range
+
+            match kind with
+            | _ when TargetCapability.subsumes target.Capability capability -> None
+            | ConditionalEqualityInOperation
+            | UnrestrictedEquality ->
+                let code =
+                    if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                        WarningCode.UnsupportedResultComparison
+                    else
+                        WarningCode.ResultComparisonNotInOperationIf
+                Some(code, [ target.Name ])
+            | ReturnInDependentBranch ->
+                if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                    None
+                else
+                    Some(WarningCode.ReturnInResultConditionedBlock, [ target.Name ])
+            | SetInDependentBranch name ->
+                if target.Capability.ResultOpacity = ResultOpacity.opaque then
+                    None
+                else
+                    Some(WarningCode.SetInResultConditionedBlock, [ name; target.Name ])
+            |> Option.map (fun (code, args) -> QsCompilerDiagnostic.Warning (code, args) range)
+
+        else
+            let range = QsNullable.defaultValue Range.Zero range
```

Comment:
Just curiosity.
Is it possible to extract the line `let range = QsNullable.defaultValue Range.Zero range` (L71 and L46) to before the `if` (before the L45)?

Reply:
Definitely possible. Updated code.